openshift/README.md

```diff
@@ -0,0 +1,116 @@
+# Deploying Debezium on OpenShift
+
+This demo shows how to deploy Debezium on a developer OpenShift instance using [Minishift tool](https://github.com/minishift/minishift).
+
+Pre-requisities
+ * [Installed and running](https://docs.openshift.org/latest/minishift/getting-started/installing.html) Minishift
+ * [Installed](https://docs.openshift.org/latest/minishift/command-ref/minishift_oc-env.html) OpenShift CLI
+
+
+## Debezium Deployment
+Albeit Debezium comes with its own set of images we are going to re-use Kafka broker and Kafka Connect images that are built and delivered as a part of project [EnMasse](https://github.com/EnMasseProject/).
+One of the componnets of the project is a [Kafka as a Service](https://github.com/EnMasseProject/barnabas/).
+It consists of entreprise grade of configuration files and images that brings Kafka on OpenShift.
+
+First we install Kafka broker and Kafka Connect templates into our OpenShift project
+```
+oc create -f https://raw.githubusercontent.com/EnMasseProject/barnabas/master/kafka-statefulsets/resources/openshift-template.yaml
+oc create -f https://raw.githubusercontent.com/EnMasseProject/barnabas/master/kafka-connect/resources/openshift-template.yaml
+oc create -f https://raw.githubusercontent.com/EnMasseProject/barnabas/master/kafka-connect/s2i/resources/openshift-template.yaml
+```
+
+Next we will create a Kafka Connect image with deployed Debezium connectors and deploy a Kafka broker cluster and Kafka Connect cluster
```

Comment:
Instead of building a new image I was hoping the connectors could be added to the existing one externally, e.g. via some mounted volume. What do you think?

Reply:
I'd disagree to this because in this case we would throw away all advantages provided by OpensShift. The barnabas team has add s2i support as a result of discussion with me.
The volumes should be used for data not for binaries as as it is oreferred to have immutable images. Just imagine you have a shared volume. Later on you modify the content but do not restart connect cluster yet. If you then increase number of replicas your cluster will contain replicas with different binaries.
In case of s2i approach - a security fix is released for Debezium.
1. Administrator triggers the binary build with new binaries
2. Openshift builds the new image and upodates the image stream
3. Connect deployment (in a sense of k8s object) automatically detects the change in imagestream and triggers either cluster rolling update or full recereate depending on confiugration
4. A patch is automatically placed in production

The volume apporch is available by barnabas too but it is intended to be used by k8s only as it lacks (currently) the s2i feature.